plugins/cpu/fu-cpu-device.c

```diff
@@ -95,12 +109,40 @@ fu_cpu_device_convert_vendor (const gchar *vendor)
 	return vendor;
 }
 
+static gboolean
+fu_cpu_is_updatable (const gchar *ucode_dir)
+{
+	g_autofree gchar *sysfs_ucode_reload = fu_common_get_path (FU_PATH_KIND_MICROCODE_RELOAD);
+
+	/* This path may not be present on older kernels and won't be supported */
+        if (g_access (sysfs_ucode_reload, R_OK|W_OK) < 0)
+                return FALSE;
+
+	if (ucode_dir != NULL) {
+		if (g_access (ucode_dir, R_OK|W_OK) < 0)
+			return FALSE;
+	} else {
+		 if (g_access ("/lib/firmware", R_OK|W_OK) < 0)
```

Comment:
So an important question - if you change `FU_PATH_KIND_FIRMWARE_SEARCH` does the `/sys/devices/system/cpu/microcode/reload` still work?  Because then that means you don't really need write access to /lib/firmware, and this could potentially work in an atomic OS as well.  You can can check if write is OK, and if not, fall back to setting that path.